### ISSUE TITLE
"The "path" argument must be of type string. Received undefined" after upgrading to 5.21.1

### ISSUE THREAD

**Issue by titusdecali** [prisma/prisma#25450]
### Bug description

Was going to give Optimize a test, but upgrading to 5.21.1 keeps causing the error below when running `prisma generate`

Can I use optimize on Prisma `5.17.0`?

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Error: 
The "path" argument must be of type string. Received undefined
```

I've reference the previous issues [6372](https://github.com/prisma/prisma/issues/6372) and [6770](https://github.com/prisma/prisma/issues/6770) already.

I'm using `5.21.1` for both `prisma` and `prisma/client` (without the carat) and I've tried wiping my `pnpm-lock.yaml` + `node_modules` folding and reinstalling. No dice. I didn't have this issue with `5.17.0`. I've also tried it on `node 18`, same result.

### How to reproduce

```
dependancies:
    "nuxt": "^3.13.2",
    "prisma": "5.21.1"
    "@prisma/extension-accelerate": "^1.2.1",
    "@prisma/extension-optimize": "^1.0.1",
devDependancies:
    "prisma": "5.21.1",
```
    

### Prisma information

```prisma
Schema worked before upgrading to 5.21.1. If you really need this for troubleshooting, I can provide it later.
```

```ts
import { PrismaClient } from '@prisma/client'
import { withOptimize } from '@prisma/extension-optimize'
import { withAccelerate } from '@prisma/extension-accelerate'
const config = useRuntimeConfig()

const prisma = new PrismaClient()
  .$extends(withAccelerate())
  .$extends(
    withOptimize({ apiKey: config.private.PRISMA_OPTIMIZE_API_KEY as string })
  )

export default function usePrisma() {
  return { prisma }
}

```

### Environment & setup

- OS: Mac Ventura 13.6
- Database: Supabase
- Node.js version: `21.0.0` / `18`

### Prisma Version

```
dependancies:
    "prisma": "5.21.1"
    "@prisma/extension-accelerate": "^1.2.1",
    "@prisma/extension-optimize": "^1.0.1",
devDependancies:
    "prisma": "5.21.1",
```

**Comment by SevInf** [comment]
Hi @titusdecali.
Thank you for opening the issue. Yes, we need also the info we asking we asking for in the issue template for troubleshooting. Please, provide your schema.

**Comment by titusdecali** [comment]
@SevInf Schema added

**Comment by SevInf** [comment]
@titusdecali I can not reproduce your issue, `prisma generate` finishes succesfully for me. Could you share an isolated repository where we could see error happening?

**Comment by titusdecali** [comment]
@SevInf I've just set up a local starter project, and I am seeing the exact same errors. I've invited you to the repo.

Also, I noticed that when using `npx prisma` and then `npx prisma init` it is adding version `"5.11.0"`. and not `"5.21.1"`
If I run it with `"5.11.0"`, it actually works as expected

Perhaps this is the cause?

**Comment by SevInf** [comment]
@titusdecali I can not reproduce the issue just from your description. I've started a project, copied your schema and executed `generate` command and it work normally for me. Could you please share the repository you just started?

**Comment by titusdecali** [comment]
Yes, I've invited you directly, but it is here: https://github.com/titusdecali/nuxt3-prisma-bug-reproduction

**Comment by SevInf** [comment]
I've checked your repository out, run `npm install`, then `npx prisma generate` and it succefully completed to me. 
`npx prisma` resolving to 5.11.0 is suspicious, it does resolve to `5.21.1` to me. Could you try with `npx prisma@5.21.1` explicitly?

**Comment by titusdecali** [comment]
I am using `pnpm`, but `npm` causes the following issue
![SHOT 19 51 51@2x](https://github.com/user-attachments/assets/58bf9b48-02b3-4a1d-9c16-1c64e3aa9b19)

I've attached the log file it referenced: 
[2024-10-18T10_45_46_792Z-debug-0.log](https://github.com/user-attachments/files/17434238/2024-10-18T10_45_46_792Z-debug-0.log)

when manually setting the version in package.json, (as `npx prisma@5.21.1` didn't seem to do that for me) I get the same error as before:

![SHOT 19 54 42@2x](https://github.com/user-attachments/assets/8e82b95d-c46d-48ca-ba12-d528dd6bdaac)
